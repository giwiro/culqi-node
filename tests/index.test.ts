import Culqi from '../src/index';
import vars from '../src/vars';

describe('Culqi', () => {
  it('should throw error if privateKey is not set', () => {
    let error;
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi();
    } catch (e) {
      error = e;
    }
    expect(error).toMatchSnapshot();
  });

  it('should set private key', () => {
    // tslint:disable-next-line:no-unused-expression
    new Culqi({
      privateKey: 'private-key-test',
    });
    expect(vars.privateKey).toBe('private-key-test');
  });

  it('should throw error if pciCompliant is true but no public key was provided', () => {
    let error;
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi({
        privateKey: 'abc',
        pciCompliant: true,
      });
    } catch (e) {
      error = e;
    }
    expect(error).toMatchSnapshot();
  });

  it('should throw error if publicKey is provided but pciCompliant is not true', () => {
    let error;
    try {
      // tslint:disable-next-line:no-unused-expression
      new Culqi({
        privateKey: 'abc',
        publicKey: 'abc',
      });
    } catch (e) {
      error = e;
    }
    expect(error).toMatchSnapshot();
  });

  it('should set privateKey and publicKey', () => {
    // tslint:disable-next-line:no-unused-expression
    new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(vars.privateKey).toBe('abc');
    expect(vars.publicKey).toBe('bcd');
  });

  it('should have tokens property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.tokens).toMatchSnapshot();
  });

  it('should have charges property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.charges).toMatchSnapshot();
  });

  it('should have refunds property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.refunds).toMatchSnapshot();
  });

  it('should have customers property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.customers).toMatchSnapshot();
  });

  it('should have cards property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.cards).toMatchSnapshot();
  });

  it('should have plans property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.plans).toMatchSnapshot();
  });

  it('should have subscriptions property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.subscriptions).toMatchSnapshot();
  });

  it('should have orders property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.orders).toMatchSnapshot();
  });

  it('should have events property', () => {
    const culqi = new Culqi({
      privateKey: 'abc',
      publicKey: 'bcd',
      pciCompliant: true,
    });
    expect(culqi.events).toMatchSnapshot();
  });
});
