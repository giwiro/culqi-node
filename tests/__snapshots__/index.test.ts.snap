// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Culqi should have cards property 1`] = `
{
  "createCard": [Function],
  "deleteCard": [Function],
  "getCard": [Function],
  "getCards": [Function],
  "updateCard": [Function],
}
`;

exports[`Culqi should have charges property 1`] = `
{
  "captureCharge": [Function],
  "createCharge": [Function],
  "getCharge": [Function],
  "getCharges": [Function],
  "updateCharge": [Function],
}
`;

exports[`Culqi should have customers property 1`] = `
{
  "createCustomer": [Function],
  "deleteCustomer": [Function],
  "getCustomer": [Function],
  "getCustomers": [Function],
  "updateCustomer": [Function],
}
`;

exports[`Culqi should have events property 1`] = `
{
  "getEvent": [Function],
  "getEvents": [Function],
}
`;

exports[`Culqi should have orders property 1`] = `
{
  "confirmOrder": [Function],
  "createOrder": [Function],
  "deleteOrder": [Function],
  "getOrder": [Function],
  "getOrders": [Function],
  "updateOrder": [Function],
}
`;

exports[`Culqi should have plans property 1`] = `
{
  "createPlan": [Function],
  "deletePlan": [Function],
  "getPlan": [Function],
  "getPlans": [Function],
  "updatePlan": [Function],
}
`;

exports[`Culqi should have refunds property 1`] = `
{
  "createRefund": [Function],
  "getRefund": [Function],
  "getRefunds": [Function],
  "updateRefunds": [Function],
}
`;

exports[`Culqi should have subscriptions property 1`] = `
{
  "createSubscription": [Function],
  "deleteSubscription": [Function],
  "getSubscription": [Function],
  "getSubscriptions": [Function],
  "updateSubscription": [Function],
}
`;

exports[`Culqi should have tokens property 1`] = `
{
  "createToken": [Function],
  "getToken": [Function],
  "getTokens": [Function],
  "updateToken": [Function],
}
`;

exports[`Culqi should throw error if pciCompliant is true but no public key was provided 1`] = `[Error: Provide 'publicKey' property]`;

exports[`Culqi should throw error if privateKey is not set 1`] = `[Error: Provide 'privateKey' property]`;

exports[`Culqi should throw error if publicKey is provided but pciCompliant is not true 1`] = `[Error: Provide 'pciCompliant' options as true. It means you acknowledge that you are PCI-compliant in order to execute certain operations.]`;
